File: src/main/java/net/guizhanss/guizhancraft/implementation/setup/ResearchSetup.java
package net.guizhanss.guizhancraft.implementation.setup;

import net.guizhanss.guizhancraft.utils.Keys;

import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack;
import io.github.thebusybiscuit.slimefun4.api.researches.Research;

import net.guizhanss.guizhancraft.implementation.GuizhanCraft;

import lombok.experimental.UtilityClass;

/**
 * This class setup all {@link Research} for GuizhanCraft
 *
 * @author ybw0014
 */
@UtilityClass
public final class ResearchSetup {

    private static int researchId = 1919810;

    public static void setup() {

    }

    public static void setupElectricSpawners() {

    }

    private static void addResearch(String key, int cost, SlimefunItemStack... items) {
        new Research(
            Keys.getResearch(key),
            ++researchId,
            GuizhanCraft.getLocalization().getResearchName(key),
            cost
        ).addItems(items).register();
    }
}

File: src/main/kotlin/net/guizhanss/guizhancraft/implementation/setup/ResearchSetup.kt
package net.guizhanss.guizhancraft.implementation.setup

import io.github.thebusybiscuit.slimefun4.api.researches.Research
import net.guizhanss.guizhancraft.GuizhanCraft
import net.guizhanss.guizhancraft.implementation.items.GCItems
import net.guizhanss.guizhancraft.utils.constants.gcKey
import net.guizhanss.guizhanlib.kt.slimefun.items.toItem

object ResearchSetup {

    val ELECTRIC_SPAWNER_ASSEMBLER = Research(
        gcKey("electric_spawner_assembler"),
        1919810,
        GuizhanCraft.localization.getString("researches.electric-spawner-assembler"),
        5
    )

    init {
        if (GuizhanCraft.integrationService.electricSpawnersEnabled) {
            ELECTRIC_SPAWNER_ASSEMBLER.addItems(
                GCItems.ELECTRIC_SPAWNER_FRAMEWORK.toItem(),
                GCItems.ELECTRIC_SPAWNER_ASSEMBLER.toItem()
            )
        }
        ELECTRIC_SPAWNER_ASSEMBLER.register()
    }
}
